h2o-core/src/main/java/water/ExternalFrameReader.java

```diff
@@ -0,0 +1,163 @@
+package water;
+
+import water.fvec.Chunk;
+import water.fvec.ChunkUtils;
+import water.fvec.Frame;
+import water.parser.BufferedString;
+
+import java.io.IOException;
+import java.nio.channels.SocketChannel;
+import java.util.UUID;
+
+/**
+ * Use to read from H2O Frames from non H2O environments
+ */
+public class ExternalFrameReader {
+
+    private static final byte IS_NA = 1;
+    private static final byte NOT_NA = 0;
+
+    // hints for expected types in order to handle download properly
+    public static final byte EXPECTED_INT = 0;
+    public static final byte EXPECTED_DOUBLE = 1;
+    public static final byte EXPECTED_STRING = 2;
+
+    private AutoBuffer ab;
+    private String keyName;
+    private byte[] expectedTypes;
```

Comment:
Actually, the reader doesn't need to pass even vector types.

They are available on the h2o backend. So When we initiate the connection, the h2o backend can use the frame name, get the frame and its vectors and convert these vectors to expected values ( detect optimal numeric values). 

So this has advantage of: 
- 1 less argument
- prepareExpectedTypes method is no h2o backend instead of in sparkling-water code ( all we need to know to create expected types is to have vector types)

In case of writer we still need to pass vector types, but again, we do not have to prepare expected types on the sparkling-water side.